sdk/clientcore/tools/sdk-codegen-tool/README.md

```diff
@@ -0,0 +1,135 @@
+# Codegen Compile time generator Plugin
+
+A Java annotation processor for generating HTTP service implementations based on annotated interfaces.
```

Comment:
We might not always do this for generating HTTP service implementations. Might be best to just say this is the client-core annotation processor, for introducing compile-time code generation for libraries based on client core?